samples/net/sockets/http_server/src/main.c

```diff
@@ -162,6 +193,8 @@ static struct http_resource_detail_dynamic uptime_resource_detail = {
 	.cb = uptime_handler,
 	.data_buffer = uptime_buf,
 	.data_buffer_len = sizeof(uptime_buf),
+	.request_headers = uptime_request_headers,
```

Comment:
With HTTP/2 we may receive individual header fields before we even get the resource URI (in HTTP/2 it's just another header field, no guaranteed order). This means that matching requested header fields at the resource level wouldn't work easily in such case, we'd need to buffer all received headers before we get the URI anyway for such a design to work.

Just thinking out loud here, as it's tricky topic, but I think that in order for this header exchange to be scalable to HTTP/2 we'd need to configure the requested headers globally in the server, instead of per individual resources? Plus, since with HTTP/2 we'd need some internal buffering anyway, perhaps we could/should revisit the memory management approach for exchanging headers with the app. While I'm generally reluctant to dynamic buffer allocation in embedded, having static buffers for headers each client context doesn't seem scalable... Maybe a dedicated heap, just for header buffering could be a good choice.

@jukkar Feel free to chime in if you have any ideas how to approach the issue.

Reply:
I would avoid dynamic memory allocation as long as possible, as then we can have memory leaks which is especially bad in embedded devices. But if we need to store data for some time, it is hard to avoid dynamic memory allocation.
One option mentioned might be the user to provide a buffer which could be populated by the server and which the headers could be copied to, and then server providing pointers to it (the argc/argv approach).